ts/features/bonus/cgn/components/merchants/CgnDiscountDetail.tsx

```diff
@@ -0,0 +1,147 @@
+    textAlign: "center",
+    backgroundColor: "#EB9505"
+  },
+  container: {
+    paddingTop: 16
+  },
+  codeContainer: { alignItems: "center", justifyContent: "space-between" },
+  codeText: {
+    fontSize: 20
+  },
+  flexEnd: { alignSelf: "flex-end" },
+  discountValue: { textAlign: "center", lineHeight: 30 }
+});
+
+const PERCENTAGE_SYMBOL = "%";
+const FEEDBACK_MS = 3000;
+
+const CATEGORY_ICON_SIZE = 22;
+const COPY_ICON_SIZE = 24;
+
+const CgnDiscountDetail: React.FunctionComponent<Props> = ({
+  discount
+}: Props) => {
+  const [isTap, setIsTap] = React.useState(false);
+  const timerRetry = React.useRef<number | undefined>(undefined);
+
+  React.useEffect(
+    () => () => {
+      clearTimeout(timerRetry.current);
+    },
+    []
+  );
+
+  const handleCopyPress = () => {
+    if (discount.discountCode) {
+      setIsTap(true);
+      clipboardSetStringWithFeedback(discount.discountCode);
+      // eslint-disable-next-line functional/immutable-data
+      timerRetry.current = setTimeout(() => setIsTap(false), FEEDBACK_MS);
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <View style={IOStyles.row}>
+        {/* TODO when available and defined the icon name should be defined through a map of category codes */}
+        <IconFont
+          name={"io-theater"}
+          size={CATEGORY_ICON_SIZE}
+          color={IOColors.bluegrey}
+        />
+        <View hspacer small />
+        <H5 weight={"SemiBold"} color={"bluegrey"}>
+          {discount.category.toLocaleUpperCase()}
+        </H5>
+      </View>
+      <View spacer />
+      <H3>{I18n.t("bonus.cgn.merchantDetail.title.description")}</H3>
+      <H4 weight={"Regular"}>{discount.description}</H4>
+      <View spacer />
+      <H3>{I18n.t("bonus.cgn.merchantDetail.title.validity")}</H3>
+      <H4 weight={"Regular"}>{discount.validityDescription}</H4>
+      {discount.discountCode !== undefined && (
+        <>
+          <View spacer small />
+          <H3>{I18n.t("bonus.cgn.merchantDetail.title.discountCode")}</H3>
+          <TouchableWithoutFeedback onPress={handleCopyPress}>
+            <View style={[IOStyles.row, styles.codeContainer]}>
+              <BaseTypography
+                weight={"Bold"}
+                color={"bluegreyDark"}
+                font={"RobotoMono"}
+                style={styles.codeText}
+              >
+                {discount.discountCode}
+              </BaseTypography>
+              <IconFont
+                name={isTap ? "io-complete" : "io-copy"}
+                size={COPY_ICON_SIZE}
+                color={IOColors.blue}
+                style={styles.flexEnd}
+              />
+            </View>
+          </TouchableWithoutFeedback>
+        </>
+      )}
+      <View spacer />
+      <H3>{I18n.t("bonus.cgn.merchantDetail.title.conditions")}</H3>
+      <H4 weight={"Regular"}>{discount.conditions}</H4>
+    </View>
+  );
+};
+
+const CgnDiscountDetailHeader: React.FunctionComponent<Props> = ({
```

Comment:
```suggestion
const CgnDiscountDetailHeader = ({
```
you could avoid this type since this component doesn't need to use children props passed via parameter.
It could be a simple function